core/server.go

```diff
@@ -119,6 +120,25 @@ func WithLogrusLogger(logger *logrus.Logger) ServerOption {
 	return &loggerOption{logger}
 }
 
+type ctxOption struct {
+	ctx context.Context
+}
+
+func (o *ctxOption) apply(params *ServerParams) {
+	params.Ctx = o.ctx
+}
+
+func WithPkgLogger(configs ...log.Config) ServerOption {
```

Comment:
I've pushed a branch [here](https://github.com/andrewemeryanz/sysl-go/tree/feature/use-pkg-logger) that shows you how I think the `pkg` logger should be integrated.